src/raftstore/store/local_read.rs

```diff
@@ -0,0 +1,446 @@
+            }
+        }
+    }
+
+    fn create_peer_status(&mut self,
+                          region: metapb::Region,
+                          peer: metapb::Peer,
+                          leader_id: u64,
+                          term: u64,
+                          applied_index_term: u64) {
+        let region_id = region.get_id();
+        self.peers_status.insert(region_id,
+                                 PeerStatus::new(region,
+                                                 peer,
+                                                 leader_id,
+                                                 term,
+                                                 applied_index_term));
+    }
+
+    fn update_peer_status(&mut self, change: PeerStatusChange) {
+        let region_id = change.region_id;
+        if let Some(peer_status) = self.peers_status.get_mut(&region_id) {
+            if let Some(leader_id) = change.leader_id {
+                peer_status.leader_id = leader_id;
+            }
+            if let Some(region) = change.region {
+                peer_status.region = region;
+            }
+            if let Some(expired_time) = change.leader_lease_expired_time {
+                peer_status.leader_lease_expired_time = expired_time;
+            }
+            if let Some(term) = change.term {
+                peer_status.term = term;
+            }
+            if let Some(applied_index_term) = change.applied_index_term {
+                peer_status.applied_index_term = applied_index_term;
+            }
+        } else {
+            panic!("peer status not found for region {} in store {}, receive unexpected peer \
+                    status update [{:?}]",
+                   region_id,
+                   self.store_id,
+                   change);
+        }
+    }
+
+    fn update_region_range(&mut self,
+                           change_type: RangeChangeType,
+                           region_id: u64,
+                           end_key: Vec<u8>) {
+        match change_type {
+            RangeChangeType::Add => {
+                if self.region_ranges.insert(end_key, region_id).is_some() {
+                    panic!("[Add] range for region [{}] is existed.", region_id);
+                }
+            }
+            RangeChangeType::Remove => {
+                if self.region_ranges.remove(&end_key).is_none() {
+                    panic!("[Remove] range for region [{}] is not exist.", region_id);
+                }
+            }
+            RangeChangeType::Override => {
+                if self.region_ranges.insert(end_key, region_id).is_none() {
+                    panic!("[Override] range for region [{}] is not exist.", region_id);
+                }
+            }
+        }
+    }
+
+    fn redirect_cmd_to_raftstore(&mut self, uuid: Uuid, req: RaftCmdRequest, cb: Callback) {
+        self.redirect_cmds.insert(uuid, cb);
+        if let Err(e) = self.sendch.send(Msg::RedirectRaftCmd {
```

Comment:
I prefer send return the RedirectRaftCmd directly, so we don't need redirect_cmds.